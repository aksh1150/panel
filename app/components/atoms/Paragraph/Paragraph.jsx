import React from 'react';
import PropTypes from 'prop-types';
import './Paragraph.scss';

export default function Paragraph({ size, fontcolour, children }) {
  return (
    <p
      className={`Paragraph ${size ? size : 'medium'} ${
        fontcolour ? fontcolour : 'royal'
      }`}
    >
      {children}
    </p>
  );
}

Paragraph.propTypes = {
  children: PropTypes.node,
  size: PropTypes.string,
  fontcolour: PropTypes.string,
};
