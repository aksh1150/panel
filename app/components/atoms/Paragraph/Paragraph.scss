@import '../../../styles/variables';
@import '../../../styles/desktop.scss';
@import '../../../styles/mobile.scss';
@import '../../../styles/mixins.scss';

.Paragraph {
  font-family: LimerickMedium;
  &.large {
    @include p-mobile;
    @include respond(sm) {
      @include p-desktop;
    }
  }
  &.medium {
    @include p2-mobile;
    @include respond(sm) {
      @include p2-desktop;
    }
  }
  &.small {
    @include p3-mobile;
    @include respond(sm) {
      @include p3-desktop;
    }
  }
  &.primary {
    color: $primary;
  }
  &.white {
    color: $base-white;
  }
  &.royal {
    color: $base-dark-grey;
  }
}
