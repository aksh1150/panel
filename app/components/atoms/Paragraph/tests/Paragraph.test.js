import React from 'react';
import { render } from 'react-testing-library';

import Paragraph from '../index';

describe('<Paragraph />', () => {
  it('should render and match the snapshot', () => {
    const children = 'this is the text';
    const {
      container: { firstChild },
    } = render(<Paragraph>{children}</Paragraph>);
    expect(firstChild).toMatchSnapshot();
  });
});
