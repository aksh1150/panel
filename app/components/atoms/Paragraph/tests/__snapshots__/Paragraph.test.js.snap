// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> should render and match the snapshot 1`] = `
<p
  class="Paragraph"
>
  this is the text
</p>
`;
